Packages/vcs/Lib/colorpicker.py

```diff
@@ -0,0 +1,215 @@
+        self.selectedActor.SetMapper(self.selectedMapper);
+        self.selectedActor.GetProperty().EdgeVisibilityOn();
+        self.selectedActor.GetProperty().SetEdgeColor(0,0,0);
+        self.selectedActor.GetProperty().SetLineWidth(3);
+        self.color = color
+        # Make sure the current color is selected
+        self.selectCell(color)
+        self.color_renderer.AddActor(self.selectedActor)
+        self.click_handler = inter.AddObserver(vtk.vtkCommand.LeftButtonReleaseEvent, self.clickEvent)
+        def noop(obj, event):
+            pass
+        self.char_handler = inter.AddObserver(vtk.vtkCommand.CharEvent, noop)
+
+    def make_current(self):
+        self.render_window.MakeCurrent()
+
+    def topRendererAtPoint(self, x, y):
+        inter = self.render_window.GetInteractor()
+
+        top = None
+
+        for renderer in collection(inter.GetRenderWindow().GetRenderers()):
+            if renderer.IsInViewport(x, y):
+                if top is None or top.GetLayer() <= renderer.GetLayer():
+                    top = renderer
+        return top
+
+    def change_map(self, state):
+        self.colormap = self.colormaps[state]
+        self.colors =  [[int(c / 100.0 * 255.0) for c in self.colormap.index[i]] for i in range(len(self.colormap.index))]
+        colorData = colors_to_scalars(self.colors)
+        self.actor.GetMapper().GetInput().GetCellData().SetScalars(colorData)
+        self.actor.GetMapper().Update()
+        self.render_window.Render()
+
+    def save(self, state):
+        if self.on_save is not None:
+            self.on_save(self.colormap, self.color)
+
+        self.close()
+
+    def cancel(self, state):
+        if self.on_cancel is not None:
+            self.on_cancel()
+
+        self.close()
+
+    def selectCell(self, cellId):
+        if cellId == -1:
+            return
+        ids = vtk.vtkIdTypeArray();
+        ids.SetNumberOfComponents(1);
+        ids.InsertNextValue(cellId);
+
+        selectionNode = vtk.vtkSelectionNode();
+        selectionNode.SetFieldType(vtk.vtkSelectionNode.CELL);
+        selectionNode.SetContentType(vtk.vtkSelectionNode.INDICES);
+        selectionNode.SetSelectionList(ids);
+
+        selection = vtk.vtkSelection();
+        selection.AddNode(selectionNode);
+
+        extractSelection = vtk.vtkExtractSelection();
+
+        extractSelection.SetInputData(0, self.actor.GetMapper().GetInput());
+        extractSelection.SetInputData(1, selection);
+
+        extractSelection.Update();
+
+        selected = vtk.vtkUnstructuredGrid();
+        selected.ShallowCopy(extractSelection.GetOutput());
+
+        self.selectedMapper.SetInputData(selected);
+        self.selectedMapper.Update()
+
+    def clickEvent(self, obj, event):
+        inter = self.render_window.GetInteractor()
+
+        x, y = inter.GetEventPosition()
+
+        renderer = self.topRendererAtPoint(x, y)
+
+        if renderer:
+            picker = vtk.vtkCellPicker()
```

Comment:
Nope, I just followed some tutorial on how to select a cell; there's never more than 255 cells, so performance hasn't been an issue. I'd be happy to redo any of the parts of colorpicker; it's all a little hacky, but I was just really tired of guessing colors.